NRLMMD-GEOIPS/data_fusion
# WORKSHOP clean up optional dependency imports

# Reviewer Checklist


# Related Issues
fixes NRLMMD-GEOIPS/geoips#338
See [this PR](https://github.com/NRLMMD-GEOIPS/geoips/pull/438) for the GeoIPS implementation of this PR. Updating other repos as necessary to reflect this new implementation. Make sure to merge that branch first before merging this one.

# Testing Instructions
Run geoips/tests/integration_tests/full_test.sh to ensure these changes have passed. They passed on my end.

# Summary
*From GEOIPS#338: 2023-07-19, Clean up optional dependencies*

Some GeoIPS readers include optional dependency statements that are required to read
certain file formats. While keeping these imports as optional is ok, we should clean up
the manner in which this is implemented. To do so, we've created a separate
geoips.testing.context_manager.py script which can handle optional imports scattered
throughout the GeoIPS codebase. This is essentially replacing our old manner of optional
dependencies with a new method that keeps things clean.

    - modified: geoips/plugins/modules/procflows/data_fusion.py
